packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -412,6 +539,72 @@ class _CupertinoAlertActionSection extends StatelessWidget {
 
   bool get _layoutActionsVertically => children.length > 2;
 
+  Widget _buildHorizontalDivider() {
+    return new Container(
+      height: 0.3,
+      color: _kButtonDividerColor,
+    );
+  }
+
+  Widget _buildVerticalDivider() {
+    return new Container(
+      width: 0.3,
+      color: _kButtonDividerColor,
+    );
+  }
+
+  Widget _buildHorizontalButtons() {
+    assert(
+      children.length <= 2,
+      'Horizontal dialog buttons can only be constructed with 2 or fewer '
+      'buttons. Actual button count: ${children.length}',
+    );
+
+    if (children.length == 1) {
+      return Row(
+        children: <Widget>[
+          new Expanded(
+            child: _buildDialogButton(children[0]),
+          ),
+        ],
+      );
+    } else {
+      return Row(
+        children: <Widget>[
+          new Expanded(
+            child: _buildDialogButton(children[0]),
+          ),
+          _buildVerticalDivider(),
+          new Expanded(
+            child: _buildDialogButton(children[1]),
+          ),
+        ],
+      );
+    }
+  }
+
+  List<Widget> _buildVerticalButtons() {
+    final List<Widget> buttonsWithDividers = <Widget>[];
+    for (int i = 0; i < children.length; ++i) {
+      buttonsWithDividers.add(
+        _buildHorizontalDivider(),
+      );
+      buttonsWithDividers.add(
+        _buildDialogButton(children[i]),
+      );
+    }
+
+    return buttonsWithDividers;
+  }
```

Comment:
here's a crazy trick for `_buildVerticalButtons`:

```dart
Iterable<Widget> _buildVerticalButtons() sync* {
  for (Widget child in children) {
    yield _buildHorizontalDivider();
    yield _buildDialogButton(child);
  }
}
```

...then at the callsite use it as `_buildVerticalButtons().toList()`.